E0116 needs to be updated to new format

From: src/test/compile-fail/E0116.rs

Error E0116 needs a span_label, an updated error text and a note, updating from:

```
error[E0116]: cannot define inherent `impl` for a type outside of the crate where the type is defined; define and implement a trait or new type instead
  --> src/test/compile-fail/E0116.rs:11:1
   |
11 | impl Vec<u8> {} //~ ERROR E0116
   | ^^^^^^^^^^^^^^^
```

To:

```
error[E0116]: cannot define inherent `impl` for a type outside of the crate where the type is defined 
  --> src/test/compile-fail/E0116.rs:11:1
   |
11 | impl Vec<u8> {} //~ ERROR E0116
   | ^^^^^^^^^^^^^^^ impl for type defined outside of crate
   |
   = note: define and implement a trait or new type instead
```

--- a/src/librustc_typeck/coherence/orphan.rs
+++ b/src/librustc_typeck/coherence/orphan.rs
@@ -33,10 +33,12 @@ struct OrphanChecker<'cx, 'tcx:'cx> {
 impl<'cx, 'tcx> OrphanChecker<'cx, 'tcx> {
     fn check_def_id(&self, item: &hir::Item, def_id: DefId) {
         if def_id.krate != LOCAL_CRATE {
-            span_err!(self.tcx.sess, item.span, E0116,
+            struct_span_err!(self.tcx.sess, item.span, E0116,
                       "cannot define inherent `impl` for a type outside of the \
-                       crate where the type is defined; define and implement \
-                       a trait or new type instead");
+                       crate where the type is defined")
+                .span_label(item.span, &format!("impl for type defined outside of crate."))
+                .span_note(item.span, &format!("define and implement a trait or new type instead"))
+                .emit();
         }
     }
 

--- a/src/test/compile-fail/E0116.rs
+++ b/src/test/compile-fail/E0116.rs
@@ -8,7 +8,10 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-impl Vec<u8> {} //~ ERROR E0116
+impl Vec<u8> {}
+//~^ ERROR E0116
+//~| NOTE impl for type defined outside of crate.
+//~| NOTE define and implement a trait or new type instead
 
 fn main() {
 }

